From 22ba6e99612fd401b8ffd15694c235a170a49194 Mon Sep 17 00:00:00 2001
From: sekaiacg <sekaiacg@gmail.com>
Date: Thu, 23 Dec 2021 18:49:38 +0800
Subject: [PATCH] language: Allow early translation

Signed-off-by: sekaiacg <sekaiacg@gmail.com>
---
 twrp.cpp |  9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/twrp.cpp b/twrp.cpp
index ecf2515..c536934 100644
--- a/twrp.cpp
+++ b/twrp.cpp
@@ -259,7 +259,7 @@ static void process_recovery_mode(twrpAdbBuFifo* adb_bu_fifo, bool skip_decrypti
 		TWFunc::Fixup_Time_On_Boot();
 
 	TWFunc::Update_Log_File();
-	DataManager::ReadSettingsFile();
+	//DataManager::ReadSettingsFile();
 
 	// Run any outstanding OpenRecoveryScript
 	std::string cacheDir = TWFunc::get_log_dir();
@@ -316,7 +316,7 @@ static void process_recovery_mode(twrpAdbBuFifo* adb_bu_fifo, bool skip_decrypti
 				if (!created)
 					LOGERR("Unable to create log directory for TWRP\n");
 			}
-			DataManager::ReadSettingsFile();
+			//DataManager::ReadSettingsFile();
 #endif
 		} else {
 			if ((DataManager::GetIntValue("tw_mount_system_ro") == 0 && sys->Check_Lifetime_Writes() == 0) || DataManager::GetIntValue("tw_mount_system_ro") == 2) {
@@ -517,6 +517,9 @@ int main(int argc, char **argv) {
 	// Load up all the resources
 	gui_loadResources();
 
+	DataManager::ReadSettingsFile();
+	PageManager::LoadLanguage(DataManager::GetStrValue("tw_language"));
+
 	std::string value;
 	static char charging = ' ';
 	static int lastVal = -1;
@@ -593,7 +596,7 @@ int main(int argc, char **argv) {
 	}
 
 	// Language
-	PageManager::LoadLanguage(DataManager::GetStrValue("tw_language"));
+	//PageManager::LoadLanguage(DataManager::GetStrValue("tw_language"));
 	GUIConsole::Translate_Now();
 
 	// Fox extra setup
--
libgit2 1.7.1

