From 8930d9d45dbeb5a3b8b8d2e78c7e624cfad7fd14 Mon Sep 17 00:00:00 2001
From: ymdzq <233900743@qq.com>
Date: Tue, 29 Aug 2023 19:58:37 +0800
Subject: [PATCH] language: Allow early translation

---
 twrp.cpp |  9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/twrp.cpp b/twrp.cpp
index c7c0756..66a72d3 100644
--- a/twrp.cpp
+++ b/twrp.cpp
@@ -225,7 +225,7 @@ static void process_recovery_mode(twrpAdbBuFifo* adb_bu_fifo, bool skip_decrypti
 		TWFunc::Fixup_Time_On_Boot();
 
 	TWFunc::Update_Log_File();
-	DataManager::ReadSettingsFile();
+	//DataManager::ReadSettingsFile();
 
 	// Run any outstanding OpenRecoveryScript
 	std::string cacheDir = TWFunc::get_log_dir();
@@ -282,7 +282,7 @@ static void process_recovery_mode(twrpAdbBuFifo* adb_bu_fifo, bool skip_decrypti
 				if (!created)
 					LOGERR("Unable to create log directory for TWRP\n");
 			}
-			DataManager::ReadSettingsFile();
+			//DataManager::ReadSettingsFile();
 #endif
 		} else {
 			if ((DataManager::GetIntValue("tw_mount_system_ro") == 0 && sys->Check_Lifetime_Writes() == 0) || DataManager::GetIntValue("tw_mount_system_ro") == 2) {
@@ -472,6 +472,9 @@ int main(int argc, char **argv) {
 	// Load up all the resources
 	gui_loadResources();
 
+	DataManager::ReadSettingsFile();
+	PageManager::LoadLanguage(DataManager::GetStrValue("tw_language"));
+
 	twrpAdbBuFifo *adb_bu_fifo = new twrpAdbBuFifo();
 	TWFunc::Clear_Bootloader_Message();
 
@@ -486,7 +489,7 @@ int main(int argc, char **argv) {
 	}
 
 	// Language
-	PageManager::LoadLanguage(DataManager::GetStrValue("tw_language"));
+	//PageManager::LoadLanguage(DataManager::GetStrValue("tw_language"));
 	GUIConsole::Translate_Now();
 
 	// Fox extra setup
--
libgit2 1.7.0

