From 979d4dba4e4636292a13974c03ec78e1895aa0e6 Mon Sep 17 00:00:00 2001
From: sekaiacg <sekaiacg@gmail.com>
Date: Wed, 29 Mar 2023 20:33:14 +0800
Subject: [PATCH] twrp: fastbootd: Always unmap super device and optimize

Signed-off-by: sekaiacg <sekaiacg@gmail.com>
---
 twrp.cpp | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/twrp.cpp b/twrp.cpp
index c53693437..d552d5f6b 100644
--- a/twrp.cpp
+++ b/twrp.cpp
@@ -500,10 +500,15 @@ int main(int argc, char **argv) {
 
 #ifdef TW_LOAD_VENDOR_MODULES
 	if (startup.Get_Fastboot_Mode()) {
-		TWPartition* ven_dlkm = PartitionManager.Find_Partition_By_Path("/vendor_dlkm");
-		PartitionManager.Prepare_Super_Volume(PartitionManager.Find_Partition_By_Path("/vendor"));
-		if(ven_dlkm) {
-			PartitionManager.Prepare_Super_Volume(ven_dlkm);
+		std::vector<std::string> prepareParts = {
+			"/system_root",
+			"/vendor",
+			"/vendor_dlkm",
+			"/odm"
+		};
+		for (auto& preparePart : prepareParts) {
+			TWPartition *part = PartitionManager.Find_Partition_By_Path(preparePart);
+			if (part) PartitionManager.Prepare_Super_Volume(part);
 		}
 	}
 #endif
-- 
2.47.1

