From 20101d6cf6a8911760d04341f55ded98c71a193b Mon Sep 17 00:00:00 2001
From: ymdzq <233900743@qq.com>
Date: Tue, 29 Aug 2023 20:00:01 +0800
Subject: [PATCH] twrp: fastbootd: Always unmap super device and optimize

---
 twrp.cpp | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/twrp.cpp b/twrp.cpp
index 66a72d3..0fb0bab 100644
--- a/twrp.cpp
+++ b/twrp.cpp
@@ -502,10 +500,15 @@ int main(int argc, char **argv) {
 
 #ifdef TW_LOAD_VENDOR_MODULES
 	if (startup.Get_Fastboot_Mode()) {
-		TWPartition* ven_dlkm = PartitionManager.Find_Partition_By_Path("/vendor_dlkm");
-		PartitionManager.Prepare_Super_Volume(PartitionManager.Find_Partition_By_Path("/vendor"));
-		if(ven_dlkm) {
-			PartitionManager.Prepare_Super_Volume(ven_dlkm);
+		std::vector<std::string> prepareParts = {
+			"/system_root",
+			"/vendor",
+			"/vendor_dlkm",
+			"/odm"
+		};
+		for (auto& preparePart : prepareParts) {
+			TWPartition *part = PartitionManager.Find_Partition_By_Path(preparePart);
+			if (part) PartitionManager.Prepare_Super_Volume(part);
 		}
 	}
 #endif
--
libgit2 1.7.1

