From 72feb84d2cd710486b0f8850814f3ef2dcac3b82 Mon Sep 17 00:00:00 2001
From: sekaiacg <sekaiacg@gmail.com>
Date: Wed, 25 Dec 2024 13:18:58 +0800
Subject: [PATCH] kernel_module_loader: Add TW_XIAOMI_TOUCH_RMMOD_GOODIX_CORE

Signed-off-by: sekaiacg <sekaiacg@gmail.com>
---
 Android.mk               | 3 +++
 kernel_module_loader.cpp | 7 +++++++
 2 files changed, 10 insertions(+)

diff --git a/Android.mk b/Android.mk
index b30f0450e..4f786e838 100644
--- a/Android.mk
+++ b/Android.mk
@@ -401,6 +401,9 @@ endif
 ifeq ($(TW_XIAOMI_TOUCH_PERMISSION_FIX), true)
 	LOCAL_CFLAGS += -DTW_XIAOMI_TOUCH_PERMISSION_FIX
 endif
+ifeq ($(TW_XIAOMI_TOUCH_RMMOD_GOODIX_CORE), true)
+	LOCAL_CFLAGS += -DTW_XIAOMI_TOUCH_RMMOD_GOODIX_CORE
+endif
 ifneq ($(TW_CUSTOM_CPU_TEMP_PATH),)
 	LOCAL_CFLAGS += -DTW_CUSTOM_CPU_TEMP_PATH=$(TW_CUSTOM_CPU_TEMP_PATH)
 endif
diff --git a/kernel_module_loader.cpp b/kernel_module_loader.cpp
index e9ec46de1..999eda08e 100644
--- a/kernel_module_loader.cpp
+++ b/kernel_module_loader.cpp
@@ -61,6 +61,13 @@ bool KernelModuleLoader::Load_Vendor_Modules() {
 	vendor_module_dirs.push_back(vendor_base_dir + gki);
 #endif
 
+#ifdef TW_XIAOMI_TOUCH_RMMOD_GOODIX_CORE
+	{
+		std::string result;
+		TWFunc::Exec_Cmd("rmmod goodix_core && insmod /vendor/lib/modules/1.1/goodix_core.ko > /dev/null 2>&1", result, false);
+	}
+#endif
+
 	TWFunc::RunFoxScript("/system/bin/beforemodules.sh", "");
 
 	Check_Loaded_Kernel_Modules();
-- 
2.49.0

