From 7d9b2649f497b282c8f7fddbf0761c1ec11eb315 Mon Sep 17 00:00:00 2001
From: sekaiacg <sekaiacg@gmail.com>
Date: Tue, 29 Aug 2023 22:00:13 +0800
Subject: [PATCH 1/2] twrp-functions: Add File_Exists function

Signed-off-by: sekaiacg <sekaiacg@gmail.com>
---
 twrp-functions.cpp | 10 ++++++++++
 twrp-functions.hpp |  1 +
 2 files changed, 11 insertions(+)

diff --git a/twrp-functions.cpp b/twrp-functions.cpp
index 29963522f..b1bb1bc1e 100755
--- a/twrp-functions.cpp
+++ b/twrp-functions.cpp
@@ -859,6 +859,16 @@ bool TWFunc::Wait_For_Battery(std::chrono::nanoseconds timeout) {
 	return false;
 }
 
+bool TWFunc::File_Exists(const string& path)
+{
+	struct stat st;
+
+	if (stat(path.c_str(), &st) == 0) {
+		return !S_ISDIR(st.st_mode);
+	}
+	return false;
+}
+
 #ifndef BUILD_TWRPTAR_MAIN
 
 // Returns "/path" from a full /path/to/file.name
diff --git a/twrp-functions.hpp b/twrp-functions.hpp
index f4a80a0bb..23f02b5bb 100755
--- a/twrp-functions.hpp
+++ b/twrp-functions.hpp
@@ -90,6 +90,7 @@ public:
 	static int32_t timespec_diff_ms(timespec& start, timespec& end);            // Returns diff in ms
 	static bool Wait_For_File(const string& path, std::chrono::nanoseconds timeout); // Wait For File, True is success, False is timeout;
 	static bool Wait_For_Battery(std::chrono::nanoseconds timeout);             // Wait For /sys/class/power_supply/battery or TW_CUSTOM_BATTERY_PATH, True is success, False is timeout;
+	static bool File_Exists(const string& path);                                // File exists
 
 #ifndef BUILD_TWRPTAR_MAIN
 	static void Write_MIUI_Install_Status(std::string install_status, bool verify);                                // Write last install status in to the /cache/recovery/last_status
-- 
2.47.1


From 87c2e518d70a61b8b6b595ebc928b77753b3e028 Mon Sep 17 00:00:00 2001
From: sekaiacg <sekaiacg@gmail.com>
Date: Fri, 13 Jan 2023 16:05:22 +0800
Subject: [PATCH 2/2] Set_FBE_Status: Add conditions to optimize judgment

Signed-off-by: sekaiacg <sekaiacg@gmail.com>
---
 partition.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/partition.cpp b/partition.cpp
index d49bf3809..b4f616429 100644
--- a/partition.cpp
+++ b/partition.cpp
@@ -781,7 +781,8 @@ void TWPartition::Set_FBE_Status() {
 	DataManager::SetValue(TW_IS_DECRYPTED, 1);
 	Is_Encrypted = true;
 	Is_Decrypted = true;
-	if (Key_Directory.empty()) {
+	PartitionManager.Mount_By_Path("/metadata", false);
+	if (Key_Directory.empty() || !TWFunc::File_Exists(Key_Directory + "/key/encrypted_key")) {
 		Is_FBE = false;
 		DataManager::SetValue(TW_IS_FBE, 0);
 	} else {
-- 
2.47.1

