From 0f6e7c0e9fd8394a992097a7f0c0ad4318e6f60d Mon Sep 17 00:00:00 2001
From: sekaiacg <sekaiacg@gmail.com>
Date: Wed, 25 Dec 2024 13:16:18 +0800
Subject: [PATCH] kernel_module_loader: Add TW_XIAOMI_TOUCH_PERMISSION_FIX

Signed-off-by: sekaiacg <sekaiacg@gmail.com>
---
 Android.mk               | 3 +++
 kernel_module_loader.cpp | 4 ++++
 2 files changed, 7 insertions(+)

diff --git a/Android.mk b/Android.mk
index 46b78a85c..b30f0450e 100644
--- a/Android.mk
+++ b/Android.mk
@@ -398,6 +398,9 @@ endif
 ifneq ($(TW_CUSTOM_BATTERY_PATH),)
 	LOCAL_CFLAGS += -DTW_CUSTOM_BATTERY_PATH=$(TW_CUSTOM_BATTERY_PATH)
 endif
+ifeq ($(TW_XIAOMI_TOUCH_PERMISSION_FIX), true)
+	LOCAL_CFLAGS += -DTW_XIAOMI_TOUCH_PERMISSION_FIX
+endif
 ifneq ($(TW_CUSTOM_CPU_TEMP_PATH),)
 	LOCAL_CFLAGS += -DTW_CUSTOM_CPU_TEMP_PATH=$(TW_CUSTOM_CPU_TEMP_PATH)
 endif
diff --git a/kernel_module_loader.cpp b/kernel_module_loader.cpp
index a8dfd8bf2..e9ec46de1 100644
--- a/kernel_module_loader.cpp
+++ b/kernel_module_loader.cpp
@@ -126,6 +126,10 @@ exit:
 	if (requested_modules_loaded == 0)
 		DataManager::SetValue("of_request_switch_control_mode" , "1");
 
+#ifdef TW_XIAOMI_TOUCH_PERMISSION_FIX
+	TWFunc::Exec_Cmd("chmod 0664 /sys/class/touch/touch_dev/touch_thp* > /dev/null 2>&1", false);
+#endif
+
 	android::base::SetProperty(TW_MODULES_MOUNTED_PROP, "true");
 
 	return true;
-- 
2.49.0

